clang detects infinite recursion

```
src/ext/yuni/src/yuni/io/filename-manipulation.cpp:302:6: warning: infinite recursion detected [-Winfinite-recursion]
src/ext/yuni/src/yuni/io/filename-manipulation.cpp:304:21: note: recursive call
  304 |     parent_path_impl(out, path, systemDependant);

```

--- a/src/ext/yuni/src/yuni/io/filename-manipulation.cpp
+++ b/src/ext/yuni/src/yuni/io/filename-manipulation.cpp
@@ -283,7 +283,7 @@ bool IsAbsolute(const AnyString& filename)
 }
 
 template<class StringT>
-static inline void parent_path_impl(StringT& out, const AnyString& path, bool systemDependant)
+static inline void parent_path_impl_static(StringT& out, const AnyString& path, bool systemDependant)
 {
     AnyString::size_type pos = (systemDependant)
                                  ? path.find_last_of(IO::Constant<char>::Separator)
@@ -296,12 +296,12 @@ static inline void parent_path_impl(StringT& out, const AnyString& path, bool sy
 
 void parent_path(String& out, const AnyString& path, bool systemDependant)
 {
-    parent_path_impl(out, path, systemDependant);
+    parent_path_impl_static(out, path, systemDependant);
 }
 
 void parent_path_impl(Clob& out, const AnyString& path, bool systemDependant)
 {
-    parent_path_impl(out, path, systemDependant);
+    parent_path_impl_static(out, path, systemDependant);
 }
 
 template<class StringT>

